R.NET/Devices/CharacterDeviceAdapter.cs

```diff
@@ -10,8 +10,18 @@ namespace RDotNet.Devices
 {
     internal class CharacterDeviceAdapter : IDisposable
     {
-        private readonly ICharacterDevice device;
+        /// <summary>
+        /// When R calls the character device (unamanged R calling managed code),
+        /// it sometimes calls the method with 'this == null' when writing/reading
+        /// from console (this seems to happen on Mono and may be a bug).
+        ///
+        /// The (somewhat incorrect) workaround is to keep the last device in a static
+        /// field and use it when 'this == null' (the check is done in 'this.Device').
+        /// This workarounds: http://rdotnet.codeplex.com/workitem/154
+        /// </summary>
+        private static ICharacterDevice lastDevice;

```

Comment:
*shrug* I'm not sure what the correct call is. @jmp75 might have an opinion. I'm going to merge this PR and test it with my custom devices and product on Windows and see what happens.